pandas/core/dtypes/missing.py

```diff
@@ -484,6 +484,20 @@ def _array_equivalent_object(left, right, strict_nan):
     return True
 
 
+def array_equals(left: ArrayLike, right: ArrayLike) -> bool:
+    """
+    ExtensionArray-compatible implementation of array_equivalent.
+    """
+    if not is_dtype_equal(left.dtype, right.dtype):
+        return False
+    elif isinstance(left, ABCExtensionArray):
+        return left.equals(right)
+    elif isinstance(right, ABCExtensionArray):
+        return False
```

Comment:
I suppose this elif block is not needed since you already checked that the dtypes are equal?

Reply:
good catch, will update